test/e2e/receive_test.go

```diff
@@ -101,51 +187,161 @@ func TestReceive(t *testing.T) {
 
 		testutil.Ok(t, q.WaitSumMetricsWithOptions(e2e.Equals(3), []string{"thanos_store_nodes_grpc_connections"}, e2e.WaitMissingMetrics))
 
-		queryAndAssertSeries(t, ctx, q.HTTPEndpoint(), queryUpWithoutInstance, promclient.QueryOptions{
+		result := instantQuery(t, ctx, q.HTTPEndpoint(), "count(up) by (prometheus)", promclient.QueryOptions{
 			Deduplicate: false,
-		}, []model.Metric{
-			{
-				"job":        "myself",
-				"prometheus": "prom1",
-				"receive":    "i2",
-				"replica":    "0",
-				"tenant_id":  "default-tenant",
-			},
-			{
-				"job":        "myself",
-				"prometheus": "prom2",
-				"receive":    "i1",
-				"replica":    "0",
-				"tenant_id":  "default-tenant",
+		}, 3)
+
+		// TODO(ianbillett): change this assertion to testutil.Equals when #4359 is fixed.
+		// Currently, we expect this e2e test to fail due to bugs in the reicever split proposal implementation.
+		for _, series := range result {
+			testutil.NotEqual(t, series.Value, 2.0)
+		}
+	})
+
+	t.Run("routing_tree", func(t *testing.T) {
+		/*
+			The routing_tree suite configures a valid and plausible, but non-trivial topology of receiver components.
+			Crucially, the first router routes to both a routing component, and a receiving component. This demonstrates
+			Receiver's ability to handle arbitrary depth receiving trees.
+
+			Router1 is configured to duplicate data twice, once to Ingestor1, and once to Router2,
+			Router2 is also configured to duplicate data twice, once to Ingestor2, and once to Ingestor3.
+
+					   ┌───────┐         ┌───────┐
+					   │       │         │       │
+					   │ Prom1 ├──┐   ┌──┤ Prom2 │
+					   │       │  │   │  │       │
+					   └───────┘  │   │  └───────┘
+							   ┌──▼───▼──┐
+							   │         │
+							   │ Router1 │
+						  ┌────┤         ├───────┐
+						  │    └─────────┘       │
+					  ┌───▼─────┐          ┌─────▼─────┐
+					  │         │          │           │
+					  │ Router2 │          │ Ingestor1 │
+				  ┌───┤         ├───┐      │           │
```

Comment:
Something looks off here

Reply:
Yeah - it renders properly in Goland :thinking: I'll have a play and see if I can make it look right.